### PR TITLE
Make Screen.Push asynchronous when necessary

### PR DESCRIPTION
**Pull request by peppy**
Also changes behaviour of LoadComponentAsync to make it more versatile.

Will be adding tests to `Screen` behaviour as it is quite important we get things right.

- [x] Depends on #1524.

### CODE REVIEW

File: osu.Framework/Graphics/Drawable.cs [hunk 1]
```diff
@@ -138,7 +138,7 @@ private void dispose(bool isDisposing)
         internal Task LoadAsync(Game game, Drawable target, Action onLoaded = null)
         {
             if (loadState != LoadState.NotLoaded)
-                throw new InvalidOperationException($@"{nameof(LoadAsync)} may not be called more than once on the same Drawable.");
```
Comment (Tom94) [inline osu.Framework/Graphics/Drawable.cs]: What happens when `loadTask` is null due to the drawable having been loaded synchronously?

Comment (peppy) [reply osu.Framework/Graphics/Drawable.cs]: What would you expect to happen here? Maybe return a completed task?

I'll add proper tests for this change and screens in general, please leave open for now.

Comment (Tom94) [reply osu.Framework/Graphics/Drawable.cs]: Perhaps returning a completed task makes most sense. Not sure how async/await normally work in such a situation, but I think we should try following the paradigms as closely as possible.

**Comment by smoogipoo** [comment]
As discussed - won't be merging this until tests are in-place.

**Comment by peppy** [comment]
Remaining test failures will be fixed once #1524 is merged.

**Pull request merged**